casanovo/denovo/model_runner.py

```diff
@@ -163,18 +163,28 @@ def log_metrics(self, test_index: AnnotatedSpectrumIndex) -> None:
             Index containing the annotated spectra used to generate model
             predictions
         """
-        model_output = [psm.sequence for psm in self.writer.psms]
-        spectrum_annotations = [
-            test_index[i][4] for i in range(test_index.n_spectra)
-        ]
+        seq_pred = []
+        seq_true = []
+        pred_idx = 0
+
+        with test_index as t_ind:
+            for true_idx in range(t_ind.n_spectra):
+                seq_true.append(t_ind[true_idx][4])
+                if pred_idx < len(self.writer.psms) and self.writer.psms[
```

Comment:
**todo:** I think this might not fully work if there are gaps of multiple skipped spectra in a row.

Reply:
I added a test case with a gap of multiple skipped spectra in the middle of the spectra annotations and one where the gap ends at the end of the spectra annotations, and everything seems to be working as expected.